tests/test-stm32-mcu-props.stanza

```diff
@@ -0,0 +1,164 @@
+#use-added-syntax(jitx)
+defpackage ocdb/tests/test-stm32-mcu-props :
+
+  import core
+  import collections
+  import math
+
+  import utils/time
+
+  import jitx
+  import jitx/commands
+
+  import ocdb/defaults
+  import ocdb/land-patterns
+  import ocdb/symbols
+  import ocdb/symbol-utils
+  import ocdb/generic-components
+
+  import ocdb/bundles
+  import ocdb/box-symbol
+  import ocdb/property-structs
+  import ocdb/generator-utils
+  import ocdb/checks
+  import ocdb/tolerance
+  import ocdb/micro-controllers
+  import ocdb/design-vars
+
+  import ocdb/db-parts 
+  import ocdb/st-microelectronics/generate-ioc
+
+public pcb-module test-stm32-mcu-props :
+  port supply : power
+
+  public net vdd (supply.vdd)
+  val num-vdd-pins = 12
+  val num-vss-pins = 9
+  public net gnd (supply.gnd)
+  net vssa (gnd)
+
+  inst mcu : ocdb/st-microelectronics/STM32F407ZET6/component
+  
+  inst cd100n0 : ceramic-cap(["capacitance" => 100.0e-9 "case" => "0603"])[12]
+  inst cd4u7 : ceramic-cap(["capacitance" => 4.7e-6 "case" => "0603"])
+  
+  inst fb : chip-resistor(["resistance" => 10.0, "case" => "0805"])
+  inst xtal : ocdb/citizen/CMJ206T32768DZBT/component
+  inst debug-conn : ocdb/tag-connect/TC2030-IDC/component
+
+
+  require debug-power : power from debug-conn
+  require debug : swd() from debug-conn
+  require conn-reset : reset from debug-conn
+
+  require lf-osc : low-freq-oscillator from mcu
+  require mcu-uart : uart() from mcu
+  require mcu-debug : swd() from mcu
+
+  ; how to count the number of VDD pins in instance mcu? punt for now
+  ; 
+  for i in 0 to num-vdd-pins do :
+    net (vdd mcu.VDD[i])
+  net (vdd mcu.PDR_ON) ; use on-board POR/PDR
+  net (gnd xtal.gnd)
+  net (supply.vdd debug-power.vdd)
+
+  for i in 0 to num-vss-pins do :
+    net (gnd mcu.VSS[i])
+  
+  public net vcap-1 (mcu.VCAP[1])
+  public net vcap-2 (mcu.VCAP[2])
+  
+  bypass-cap-strap(mcu.VCAP[1], gnd, 2.2e-6)
+  bypass-cap-strap(mcu.VCAP[2], gnd, 2.2e-6)
+  
+  res-strap(mcu.BOOT[0], gnd, 1.0e3)
+  
+  public net vdda (mcu.VDDA mcu.VREF+)
+    
+  bypass-cap-strap(mcu.VDDA, gnd, 1.0e-6)
+  bypass-cap-strap(mcu.VDDA, gnd, 100.0e-9)
+  bypass-cap-strap(mcu.VREF+, gnd, 1.0e-6)
+  bypass-cap-strap(mcu.VREF+, gnd, 100.0e-9)
+  
+  net (vdd fb.p[1])
+  net (vdda fb.p[2])
+  
+  for i in 0 to num-vdd-pins do :
+    net (cd100n0[i].p[1] vdd)
+    net (cd100n0[i].p[2] gnd)
+    
+  net (cd4u7.p[1] vdd)
+  net (cd4u7.p[2] gnd)
+    
+  net (vssa mcu.VSSA)
+  
+  val lf-cb = add-xtal-caps(xtal, gnd, 0.5e-12)
+  
+  net (lf-osc.in xtal.p[1])
+  net (lf-osc.out xtal.p[2])
+
+  net (debug mcu-debug)
+
+  net (mcu-debug.swdio mcu-debug.swdclk)
+
+  net dummy ()
```

Comment:
We have an underused JITX concept called `node` that just names something you can net to.
```
node dummy : pin
```

Reply:
Of course a dummy net works too.

Reply:
I just use the dummy net so that the pins remain connected. Would I use your node definition above as a net in the net statement?

node dummy : pin

net (dummy mcu.somepin)

Reply:
Yes, it would work just like that!